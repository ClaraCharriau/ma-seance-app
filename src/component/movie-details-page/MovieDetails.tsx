import { useLoaderData } from 'react-router-dom';
import { Movie } from '../../models/Movie';
import style from './MovieDetails.module.css';
import useDurationFormat from '../../hooks/useDurationFormat';
import FavoriteButton from '../common/favorite-button/FavoriteButton';
import useStringListFormat from '../../hooks/useStringListFormat';
import useDateDMYFormat from '../../hooks/dates/useDateDMYFormat';

const MovieDetails = () => {
    const movie = useLoaderData() as Movie;

    const TMDB_PATH = 'https://image.tmdb.org/t/p/w500';

    const formattedDuration = useDurationFormat(movie.duration);
    const formattedReleaseDate = useDateDMYFormat(movie.releaseDate);
    const directors = useStringListFormat(movie.directors);
    const cast = useStringListFormat(movie.cast);
    const genres = useStringListFormat(movie.genres);

    return (
        <>
            <div className={style.backgroundWrapper}>
                <img
                    className={style.movieBackground}
                    src={TMDB_PATH + movie.photoLink}
                    alt={'photo de ' + movie.title}
                />
                <div className={style.fading}></div>
            </div>

            <main className={style.movieDetailsMain}>
                <section className={style.movieInformations}>
                    <FavoriteButton itemId={movie.id} itemType={'movie'} />
                    <img
                        src={TMDB_PATH + movie.posterLink}
                        alt={'affiche de ' + movie.title}
                        className={style.moviePoster}
                    />
                    <div className={style.movieInfos}>
                        <div>
                            <h2 className={style.movieTitle}>{movie.title}</h2>
                            <p className={style.movieDetails}>
                                Sorti le {formattedReleaseDate} ● {genres} ● {formattedDuration}
                            </p>

                            <p className={style.directorText}>Réalisé par {directors}</p>
                            <p className={style.castText}>Avec {cast}...</p>
                            <p className={style.resume}>{movie.resume}</p>
                        </div>
                        <a href={movie.trailerLink} className={style.trailerLink} target="_blank" rel="noreferrer">
                            <svg
                                width="28"
                                height="20"
                                viewBox="0 0 28 20"
                                fill="none"
                                xmlns="http://www.w3.org/2000/svg"
                            >
                                <path
                                    d="M26.9314 3.24972L26.9314 3.25001C27.2093 4.30825 27.3543 5.98 27.4275 7.42249C27.4638 8.13638 27.4819 8.78174 27.491 9.24865C27.4955 9.482 27.4977 9.67051 27.4989 9.8004C27.4994 9.86533 27.4997 9.9156 27.4999 9.94948L27.5 9.98777L27.5 9.99728L27.5 9.99957V10.0001V10.0002V10.0003V10.0008L27.5 10.003L27.5 10.0126L27.4999 10.0509C27.4997 10.0847 27.4994 10.135 27.4989 10.1999C27.4977 10.3298 27.4955 10.5183 27.491 10.7517C27.4819 11.2186 27.4638 11.8639 27.4275 12.5778C27.3543 14.0202 27.2093 15.6919 26.9315 16.75L26.9314 16.7503C26.6537 17.8104 25.8365 18.6403 24.8081 18.9207L24.8078 18.9208C24.3147 19.0556 23.4038 19.1653 22.2719 19.2496C21.1544 19.3327 19.8633 19.3884 18.6358 19.4256C17.409 19.4628 16.2497 19.4814 15.3968 19.4907C14.9705 19.4954 14.6209 19.4977 14.3781 19.4988C14.2567 19.4994 14.162 19.4997 14.0978 19.4999L14.0247 19.5L14.0061 19.5L14.0015 19.5H14.0004H14.0001L14 20L14 19.5H13.9997H13.9986L13.9939 19.5L13.9754 19.5L13.9023 19.4999C13.838 19.4997 13.7433 19.4994 13.6219 19.4988C13.3791 19.4977 13.0296 19.4954 12.6033 19.4907C11.7504 19.4814 10.591 19.4628 9.36427 19.4256C8.13667 19.3884 6.84564 19.3327 5.72808 19.2496C4.59624 19.1653 3.68528 19.0556 3.1922 18.9208L3.19193 18.9207C2.16369 18.6403 1.34631 17.8104 1.06868 16.7503L1.0686 16.75C0.790763 15.6918 0.645693 14.0202 0.572481 12.5778C0.536249 11.8639 0.518114 11.2186 0.509046 10.7517C0.504514 10.5183 0.502252 10.3298 0.501124 10.1999C0.500559 10.135 0.500278 10.0847 0.500139 10.0509L0.500017 10.0126L0.500002 10.003L0.5 10.0008L0.5 10.0003V10.0002V10.0002V10.0001L0.5 9.99957L0.500002 9.99728L0.500017 9.98777L0.500139 9.94948C0.500278 9.9156 0.500559 9.86534 0.501124 9.8004C0.502252 9.67051 0.504514 9.482 0.509046 9.24865C0.518114 8.78174 0.536249 8.13638 0.572481 7.4225C0.645693 5.98 0.790763 4.30826 1.0686 3.25002L1.06868 3.24972C1.3463 2.18964 2.16364 1.35997 3.19195 1.07947L3.19224 1.07939C3.68531 0.944578 4.59625 0.834777 5.7281 0.750521C6.84565 0.667329 8.13668 0.611624 9.36427 0.574408C10.591 0.537216 11.7504 0.518607 12.6033 0.5093C13.0296 0.504648 13.3791 0.502322 13.6219 0.501161C13.7433 0.50058 13.838 0.500289 13.9023 0.500144L13.9754 0.500018L13.9939 0.500002L13.9986 0.5L13.9997 0.5L14 0.5L14 0L14.0001 0.5L14.0004 0.5L14.0015 0.5L14.0061 0.500002L14.0247 0.500018L14.0978 0.500144C14.162 0.500289 14.2567 0.50058 14.3781 0.501161C14.6209 0.502322 14.9705 0.504648 15.3968 0.5093C16.2497 0.518607 17.409 0.537216 18.6358 0.574408C19.8633 0.611624 21.1544 0.667329 22.2719 0.750521C23.4038 0.834777 24.3147 0.944578 24.8078 1.07939L24.8081 1.07948C25.8366 1.35998 26.6538 2.18961 26.9314 3.24972ZM10.6999 14.2859V15.1608L11.4537 14.7167L18.7282 10.431L19.4594 10.0002L18.7282 9.56937L11.4537 5.28355L10.6999 4.83944V5.71434V14.2859Z"
                                    stroke="#FFF1A7"
                                />
                            </svg>
                            Bande annonce
                        </a>
                    </div>
                </section>
            </main>
        </>
    );
};

export default MovieDetails;
