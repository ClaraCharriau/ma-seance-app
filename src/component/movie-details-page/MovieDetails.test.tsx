import { render } from '@testing-library/react';
import mockMovie from '../../mocks/movies/movie-1.json';
import MovieDetails from './MovieDetails';

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useLoaderData: () => {
        return mockMovie;
    }
}));

describe('Movie Details Component', () => {
    it('renders movie details component', () => {
        const component = render(<MovieDetails />);

        expect(component.container).toMatchSnapshot();
    });
});
