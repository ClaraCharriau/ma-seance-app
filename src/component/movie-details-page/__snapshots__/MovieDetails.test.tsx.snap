// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Details Component renders movie details component 1`] = `
<div>
  <div
    class="backgroundWrapper"
  >
    <img
      alt="photo de Pauvres créatures"
      class="movieBackground"
      src="https://image.tmdb.org/t/p/w500/bQS43HSLZzMjZkcHJz4fGc7fNdz.jpg"
    />
    <div
      class="fading"
    />
  </div>
  <main
    class="movieDetailsMain"
  >
    <section
      class="movieInformations"
    >
      <button
        class="favoriteButtonWrapper"
        data-testid="button"
        id="favorite-button"
      >
        <svg
          class="heart"
          viewBox="0 0 33 28"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M30.0569 3.26386L30.0588 3.26597C32.9452 6.45492 32.7315 11.7267 29.5079 14.9865L29.5071 14.9872L17.7552 26.9219L17.7534 26.9237C17.5851 27.0964 17.3845 27.2337 17.1633 27.3278C16.9422 27.4218 16.7048 27.4708 16.4648 27.472C16.2249 27.4732 15.987 27.4265 15.765 27.3347C15.5429 27.2428 15.3409 27.1074 15.1709 26.9364L15.1691 26.9346L2.99284 14.8095L2.99236 14.809C2.19548 14.0176 1.56732 13.0709 1.14646 12.0267C0.725595 10.9825 0.520911 9.86282 0.544994 8.73623C0.569076 7.60964 0.821415 6.49993 1.28649 5.47497C1.75157 4.45001 2.41956 3.53143 3.24949 2.77531L3.24949 2.77532L3.25121 2.77373C6.41871 -0.139405 11.6557 0.087266 14.8974 3.32917L14.8974 3.32919L14.9006 3.33229L15.9994 4.41195L16.3538 4.76016L16.7043 4.40797L18.0972 3.00815L18.0981 3.00717C18.8883 2.2087 19.8304 1.57901 20.8681 1.1557C21.9055 0.732453 23.0172 0.524181 24.1364 0.543286C25.2556 0.564788 26.3585 0.817247 27.3768 1.2851C28.3953 1.75303 29.3075 2.42649 30.0569 3.26386Z"
          />
        </svg>
        <p
          class="favoriteButtonText"
        >
          Ce 
          film
           
          n’est pas
           dans
           
          votre watchlist
        </p>
      </button>
      <img
        alt="affiche de Pauvres créatures"
        class="moviePoster"
        src="https://image.tmdb.org/t/p/w500/5fT98da9ccWN2xr8VOJrSBp3Cdw.jpg"
      />
      <div
        class="movieInfos"
      >
        <div>
          <h2
            class="movieTitle"
          >
            Pauvres créatures
          </h2>
          <p
            class="movieDetails"
          >
            Sorti le 
            17 janvier 2024
             ● 
            Science-Fiction & Comédie
             ● 
            1h30
          </p>
          <p
            class="directorText"
          >
            Réalisé par 
            Yorgos Lanthimos
          </p>
          <p
            class="castText"
          >
            Avec 
            Emma Stone, Mark Ruffalo, Willem Dafoe
            ...
          </p>
          <p
            class="resume"
          >
            Après s'être noyée pour échapper à son mari violent, le cerveau de Bella Baxter est remplacé par celui de son enfant à naître.
          </p>
        </div>
        <a
          class="trailerLink"
          href="https://www.youtube.com/watch?v=ZGwOzkF6HjI&t=1s"
          rel="noreferrer"
          target="_blank"
        >
          <svg
            fill="none"
            height="20"
            viewBox="0 0 28 20"
            width="28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M26.9314 3.24972L26.9314 3.25001C27.2093 4.30825 27.3543 5.98 27.4275 7.42249C27.4638 8.13638 27.4819 8.78174 27.491 9.24865C27.4955 9.482 27.4977 9.67051 27.4989 9.8004C27.4994 9.86533 27.4997 9.9156 27.4999 9.94948L27.5 9.98777L27.5 9.99728L27.5 9.99957V10.0001V10.0002V10.0003V10.0008L27.5 10.003L27.5 10.0126L27.4999 10.0509C27.4997 10.0847 27.4994 10.135 27.4989 10.1999C27.4977 10.3298 27.4955 10.5183 27.491 10.7517C27.4819 11.2186 27.4638 11.8639 27.4275 12.5778C27.3543 14.0202 27.2093 15.6919 26.9315 16.75L26.9314 16.7503C26.6537 17.8104 25.8365 18.6403 24.8081 18.9207L24.8078 18.9208C24.3147 19.0556 23.4038 19.1653 22.2719 19.2496C21.1544 19.3327 19.8633 19.3884 18.6358 19.4256C17.409 19.4628 16.2497 19.4814 15.3968 19.4907C14.9705 19.4954 14.6209 19.4977 14.3781 19.4988C14.2567 19.4994 14.162 19.4997 14.0978 19.4999L14.0247 19.5L14.0061 19.5L14.0015 19.5H14.0004H14.0001L14 20L14 19.5H13.9997H13.9986L13.9939 19.5L13.9754 19.5L13.9023 19.4999C13.838 19.4997 13.7433 19.4994 13.6219 19.4988C13.3791 19.4977 13.0296 19.4954 12.6033 19.4907C11.7504 19.4814 10.591 19.4628 9.36427 19.4256C8.13667 19.3884 6.84564 19.3327 5.72808 19.2496C4.59624 19.1653 3.68528 19.0556 3.1922 18.9208L3.19193 18.9207C2.16369 18.6403 1.34631 17.8104 1.06868 16.7503L1.0686 16.75C0.790763 15.6918 0.645693 14.0202 0.572481 12.5778C0.536249 11.8639 0.518114 11.2186 0.509046 10.7517C0.504514 10.5183 0.502252 10.3298 0.501124 10.1999C0.500559 10.135 0.500278 10.0847 0.500139 10.0509L0.500017 10.0126L0.500002 10.003L0.5 10.0008L0.5 10.0003V10.0002V10.0002V10.0001L0.5 9.99957L0.500002 9.99728L0.500017 9.98777L0.500139 9.94948C0.500278 9.9156 0.500559 9.86534 0.501124 9.8004C0.502252 9.67051 0.504514 9.482 0.509046 9.24865C0.518114 8.78174 0.536249 8.13638 0.572481 7.4225C0.645693 5.98 0.790763 4.30826 1.0686 3.25002L1.06868 3.24972C1.3463 2.18964 2.16364 1.35997 3.19195 1.07947L3.19224 1.07939C3.68531 0.944578 4.59625 0.834777 5.7281 0.750521C6.84565 0.667329 8.13668 0.611624 9.36427 0.574408C10.591 0.537216 11.7504 0.518607 12.6033 0.5093C13.0296 0.504648 13.3791 0.502322 13.6219 0.501161C13.7433 0.50058 13.838 0.500289 13.9023 0.500144L13.9754 0.500018L13.9939 0.500002L13.9986 0.5L13.9997 0.5L14 0.5L14 0L14.0001 0.5L14.0004 0.5L14.0015 0.5L14.0061 0.500002L14.0247 0.500018L14.0978 0.500144C14.162 0.500289 14.2567 0.50058 14.3781 0.501161C14.6209 0.502322 14.9705 0.504648 15.3968 0.5093C16.2497 0.518607 17.409 0.537216 18.6358 0.574408C19.8633 0.611624 21.1544 0.667329 22.2719 0.750521C23.4038 0.834777 24.3147 0.944578 24.8078 1.07939L24.8081 1.07948C25.8366 1.35998 26.6538 2.18961 26.9314 3.24972ZM10.6999 14.2859V15.1608L11.4537 14.7167L18.7282 10.431L19.4594 10.0002L18.7282 9.56937L11.4537 5.28355L10.6999 4.83944V5.71434V14.2859Z"
              stroke="#FFF1A7"
            />
          </svg>
          Bande annonce
        </a>
      </div>
    </section>
  </main>
</div>
`;
